zeppelin-interpreter/src/main/java/org/apache/zeppelin/scheduler/Job.java

```diff
@@ -200,7 +203,11 @@ private synchronized void completeWithSuccess(Object result) {
   }
 
   private synchronized void completeWithError(Throwable error) {
-    setResult(error.getMessage());
+    // TODO(jl): Remove this trick. Most of error are covered by InterpreterException
+    if (error instanceof InterpreterException && null != error.getCause()) {
+      error = error.getCause();
+    }
+    setResult(new InterpreterResult(Code.ERROR, getStack(error)));
```

Comment:
do we need to return the whole stack as the result? that seems a bit much to the user. how about just the message as before and log the exception?

Reply:
I'm trying to check if `getMessage()` is enough or not.

Reply:
BTW, AFAIK, our past behaviours was also showing all stacktraces.

Reply:
I would suggest to use full stacktrace for now, as this is the behavior we used before. Although I think we should not output the full stacktrace, we need to investigate it more to make sure the error message is clear and sufficient for users. Job is one place, there's other places like (e.g. when invalid interpreter is specified )

Reply:
![image](https://user-images.githubusercontent.com/3612566/39416170-d9bce860-4c85-11e8-872c-0b450bd79b76.png)

It think it's enough for the simple error but not sure for other cases. And it would be better to keep it this time because this PR is kind of howfixes :-)